patches/minecraft/net/minecraft/item/Item.java.patch

```diff
@@ -635,13 +635,25 @@
 +        }
 +        return builder.build();
 +    }
-+
++    /**
++     * Called from EntityItemFram.getAnalogOutput, used to allow items in item frames to provide their own
++     * Comparator output signal strengths, e.g. displaying any sort of world information, such as biome
++     * temperature, rainfall, etc.
++     * @param worldIn
++     * @param pos
++     * @param stack
++     * @param rotation - the default value from the ItemFrame (rotation amount)
++     * @return redstone signal strength
++     */
++    public int getItemFrameOutput(World worldIn, BlockPos pos, ItemStack stack, int rotation) {
```

Comment:
@mcenderdragon  no, either have World/Pos or the Entity. Having both is redundant and annoying

an entity's world is always public as is its position. posX/Y/Z are more precise than BlockPos as well

The conventions for World/Pos in other methods are because those describe if the item is used on a specific block position in the world, etc. etc. If there is an entity involved just pass the entity